AndroidSDKCore/src/main/java/com/leanplum/internal/RequestOld.java

```diff
@@ -734,10 +734,21 @@ static void deleteSentRequests(int requestsCount) {
       editor.remove(Constants.Defaults.UUID_KEY);
       SharedPreferencesUtil.commitChanges(editor);
     }
-
+    // if we send less than 100% of requests, we need to reset the batch
+    // UUID for the next batch
+    if (fraction < 1) {
+      setNewBatchUUID(requestData);
+    }
     return requestData;
   }
 
+  private void setNewBatchUUID(List<Map<String, Object>> requests) {
+    String uuid = UUID.randomUUID().toString();
+    for (Map<String, Object> request : requests) {
+      request.put(UUID_KEY, uuid);
```

Comment:
Dont we need to set the new uui_key in SharedPrefs? Also, when we put in "Request", can you please help me understand, does it mean we will make that many individual calls?
I thought we had batching before making api calls.

Reply:
Lemme check if we need to set the new key in SharedPrefs. 
Putting something in Request doesn't mean it will be an individual call. It is all batched still (which is why this UUID must be set in the first place)

Reply:
@hrishileanplum I saved the new UUID in shared prefs as well now. Do pls lmk if its clear now